examples/imagenet_d_helper/map_classes.py

```diff
@@ -0,0 +1,456 @@
+import glob
+import os
+import re
+
+import numpy as np
+import torch
+
+
+def build_map_dict():
+    map_dict = {}
+    with open("examples/imagenet_d_helper/imagenet_classes.csv", "r") as r:
+        for line in r:
+            splits = line.split("\001")
+            map_dict[int(splits[0])] = splits[1]
+    return map_dict
+
+
+def get_imagenet_visda_mapping(visda_dir, map_dict):
+    # visda2imagenet
+    matching_names = dict()
+    matching_labels = dict()
+    map_dict_visda = dict()
+
+    label = 0
+    visda = os.listdir(visda_dir)
+    for label, item in enumerate(sorted(visda)):
+        map_dict_visda[item] = label
+        item_split = item.split("_")
+        for ii in item_split:
+            for j in map_dict:
+                if re.search(r"\b" + ii + r"\b", map_dict[j]):
+                    if item not in matching_names:
+                        matching_names[item] = list()
+                        matching_labels[str(label)] = list()
+
+                    matching_names[item].append([map_dict[j]])
+                    matching_labels[str(label)].append(j)
+
+    matching_names, matching_labels = clean_dataset(
+        matching_names, matching_labels, map_dict_visda, map_dict
+    )
+
+    return matching_names, matching_labels
+
+
+def create_symlinks_and_get_imagenet_visda_mapping(
+    visda_location, symlinks_location, map_dict
+):
+
+    # initial mapping and cleaning
+    matching_names, matching_labels = get_imagenet_visda_mapping(
+        visda_location, map_dict
+    )
+
+    # some classes are ambiguous
+    ambiguous_matching = get_ambiguous_classes(matching_names)
+
+    # create symlinks for all valid classes
+    symlinks_location = os.path.abspath(symlinks_location)
+    if not os.path.exists(symlinks_location):
+        os.makedirs(symlinks_location)
+    else:
+        print("Path ", symlinks_location, " exists, skipping.")
+    for folder in matching_names.keys():
+        target_folder_class = os.path.join(
+            symlinks_location, ambiguous_matching[folder]
+        )
+        if not os.path.exists(target_folder_class):
+            os.makedirs(target_folder_class)
+        try:
+            allFiles_path_jpg = visda_location + "/" + folder + "/*.jpg"
+            allFiles_path_png = visda_location + "/" + folder + "/*.png"
+            allFiles_jpg = glob.glob(allFiles_path_jpg)
+            allFiles_png = glob.glob(allFiles_path_png)
+            allFiles = allFiles_jpg + allFiles_png
+            for file in allFiles:
+                newFile = target_folder_class + "/" + file.split("/")[-1]
```

Comment:
I am having a problem with this line: 

`Cannot find the path '...\\aircraft_carrier\\painting_001_000001.jpg' -> '...\\visda_symlinks\\painting\\aircraft_carrier/aircraft_carrier\\painting_001_000001.jpg'`

I think the problem caused by `file.split("/")[-1]` and backward slash.

My suggestion is to change it to smth like this:
```suggestion
                newFile = os.path.join(target_folder_class, os.path.normpath(os.path.basename(file)))
```